### ISSUE TITLE
Different prompts per provider doesn't work

### ISSUE THREAD

**Issue by albertlieyingadrian** [promptfoo/promptfoo#1738]
I tried to load different prompts for different providers following this instruction https://www.promptfoo.dev/docs/configuration/parameters/#different-prompts-per-model

```yaml
prompts:
  - id: file://prompts/gpt_chat_prompt.json
    label: gpt_chat_prompt
  - id: file://prompts/llama_completion_prompt.txt
    label: llama_completion_prompt

providers:
  - id: openai:gpt-4o-mini
    prompts:
      - gpt_chat_prompt
  - id: openai:gpt-4o
    prompts:
      - gpt_chat_prompt
  - id: replicate:meta/meta-llama-3.1-405b-instruct
    label: llama-3.1-405b-instruct
    prompts:
      - llama_completion_prompt
```

I got an error by `invariant`
```shell
/Users/albertlieadrian/Documents/Random/ai-llm-playground/evals/node_modules/tiny-invariant/dist/tiny-invariant.cjs.js:14
    throw new Error(value);
          ^

Error: Invariant failed: Invalid prompt
```

I traced it back and I figured it was caused by this checking https://github.com/promptfoo/promptfoo/blob/main/src/config.ts#L339-L342

The prompt's data structure, in this case, would have no `raw`, only `id` and `label`. Hence this configuration will never be able to be executed by promptfoo.

**Comment by albertlieyingadrian** [comment]
This is super hacky way to make it work -- by adding `raw` field in the config
```yaml
prompts:
  - id: file://prompts/gpt_chat_prompt.json
    label: gpt_chat_prompt
  - id: file://prompts/llama_completion_prompt.txt
    label: llama_completion_prompt

providers:
  - id: openai:gpt-4o-mini
    prompts:
      - gpt_chat_prompt
  - id: openai:gpt-4o
    prompts:
      - gpt_chat_prompt
  - id: replicate:meta/meta-llama-3.1-405b-instruct
    label: llama-3.1-405b-instruct
    prompts:
      - llama_completion_prompt
```

But seems like it's not intuitive for users and this might be a blocker for them to run even basic evals if they are gonna involve different prompts per provider @mldangelo @typpo

**Comment by bfdykstra** [comment]
I also ran in to this, but only when attempting to pass a different configuration file from command line like: `npx promptfoo@latest eval --config eval_all_config.yaml`. If you include the `raw` field in the prompts, the eval runs, but with the content of the `raw` field as the prompt, not what is specified in the file

**Comment by md2eoseo** [comment]
> I also ran in to this, but only when attempting to pass a different configuration file from command line like: `npx promptfoo@latest eval --config eval_all_config.yaml`. If you include the `raw` field in the prompts, the eval runs, but with the content of the `raw` field as the prompt, not what is specified in the file

exact same issue for me

**Comment by typpo** [comment]
Fix in #1757.  A temp workaround is to run `promptfoo eval` in the same directory as the config file, without the `-c` arg

**Comment by mldangelo** [comment]
Hi @albertlieyingadrian, thank you for reporting this issue. This should be fixed in the latest release: [0.90.3](https://github.com/promptfoo/promptfoo/releases/tag/0.90.3) Please let us know if you encounter any further issues.
